testcases/cve/cve-2017-17052.c

```diff
@@ -101,7 +103,19 @@ static void run(void)
 			usleep(EXEC_USEC);
 			*do_exit = 1;
 		}
-		tst_res(TINFO, "run %d passed", run);
+
+		SAFE_WAIT(&status);
+		if (WIFEXITED(status)) {
+			if (WEXITSTATUS(status) == 0) {
+				tst_res(TINFO, "run %d passed", run);
+			} else {
+				tst_res(TFAIL, "child exited with status: %d",
+				        WEXITSTATUS(status));
+			}
+		} else {
+			tst_res(TFAIL, "child did not exit normally");
+		}
+
```

Comment:
We do have tst_strstatus() make use of it.

Reply:
Ah thanks, yes that's better. I've updated this commit.